// Copyright (c) 2023-present Hutech University. All rights reserved
// Licensed under the MIT License

syntax = "proto3";

import "google/protobuf/timestamp.proto";

service ChatHistoryService {
    rpc GetChatHistory (GetChatHistoryRequest) returns (GetChatHistoryResponse) {}
    rpc AddChatHistory (AddChatHistoryRequest) returns (AddChatHistoryResponse) {}
    rpc DeleteChatHistory (DeleteChatHistoryRequest) returns (DeleteChatHistoryResponse) {}
}

message GetChatHistoryRequest {
    string session_id = 1;
}

message GetChatHistoryResponse {
    repeated ChatHistory chat_history = 1;
}

message AddChatHistoryRequest {
    string question = 1;
    string answer = 2;
    string session_id = 3;
}

message AddChatHistoryResponse {
    ChatHistory chat_history = 1;
}

message DeleteChatHistoryRequest {
    string id = 1;
}

message DeleteChatHistoryResponse {
    string message = 1;
}

message ChatHistory {
    string id = 1;
    google.protobuf.Timestamp created_at = 2;
    string question = 3;
    string answer = 4;
    string session_id = 5;
}
